src/components/remote_control/src/resource_allocation_manager_impl.cc

```diff
@@ -70,31 +87,70 @@ AcquireResult::eType ResourceAllocationManagerImpl::AcquireResource(
 }
 
 void ResourceAllocationManagerImpl::SetAccessMode(
-    AccessMode::eType access_mode) {
+    const AccessMode::eType access_mode) {
   current_access_mode_ = access_mode;
 }
 
 void ResourceAllocationManagerImpl::AskDriver(const std::string& module_type,
-                                              uint32_t app_id,
+                                              const uint32_t hmi_app_id,
                                               AskDriverCallBackPtr callback) {
   LOG4CXX_AUTO_TRACE(logger_);
-  // Send GetInteriorConsent
-  // subscribe on GetInteriorConsent response
-  // execute callback on response
-  // event_dispatcher_.add_observer(function_id, msg[kId].asInt(), callback);
+  // Create GetInteriorConsent request to HMI
+  Json::Value params;
+  params[json_keys::kAppId] = hmi_app_id;
+  params[message_params::kModuleType] = module_type;
+  params[json_keys::kId] = rc_plugin_.service()->GetNextCorrelationID();
+  params[json_keys::kJsonrpc] = "2.0";
+  params[json_keys::kMethod] = functional_modules::hmi_api::get_user_consent;
+
+  Json::FastWriter writer;
+  application_manager::MessagePtr message_to_send(
+      new application_manager::Message(
+          protocol_handler::MessagePriority::kDefault));
+  message_to_send->set_protocol_version(
+      application_manager::ProtocolVersion::kHMI);
+  message_to_send->set_correlation_id(params[json_keys::kId].asInt());
+  std::string json_msg = writer.write(params);
+  message_to_send->set_json_message(json_msg);
+  message_to_send->set_message_type(application_manager::MessageType::kRequest);
+  LOG4CXX_DEBUG(logger_, "\nRequest to HMI: \n" << json_msg);
+  // Send GetInteriorConsent request to HMI
+  rc_plugin_.service()->SendMessageToHMI(message_to_send);
+
+  // Execute callback on response
+  rc_plugin_.event_dispatcher().add_observer(
+      params[json_keys::kMethod].asString(),
+      params[json_keys::kId].asInt(),
+      callback.get());
   active_call_back_ = callback;
 }
 
 ResourceAllocationManagerImpl::~ResourceAllocationManagerImpl() {}
 
 void ResourceAllocationManagerImpl::ForceAcquireResource(
-    const std::string& module_type, uint32_t app_id) {
+    const std::string& module_type, const uint32_t app_id) {
   LOG4CXX_DEBUG(logger_, "Force " << app_id << " acquiring " << module_type);
   allocated_resources_[module_type] = app_id;
 }
 
+bool ResourceAllocationManagerImpl::IsModuleTypeRejected(
+    const std::string& module_type, const uint32_t app_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  RejectedResources::iterator it =
```

Comment:
@LuxoftAKutsan First parameter of `in_range` is container. Not `pair` . 
- First - I consider that this way is more readable 
- Second - I don't want to use *rejected_resources_list_it->second